include/cantera/thermo/ThermoPhase.h

```diff
@@ -1556,6 +1556,13 @@ class ThermoPhase : public Phase
             const Composition& oxComp, ThermoBasis basis=ThermoBasis::molar) const;
     //! @}
 
+    //! Return common parameters used by an equation of state that a user may
+    //! want to access.
+    virtual AnyMap getEosParameters()
+    {
+        throw NotImplementedError("ThermoPhase::getEosParameters");
+    }
```

Comment:
I think the base class can just return an empty `AnyMap` rather than throwing an exception. Then, derived classes can call the base class method first to get an `AnyMap` with these parameters from the base class and add any specific to the derived class.